src/game/Player.cpp

```diff
@@ -15841,8 +15841,9 @@ void Player::SaveToDB()
         _SaveStats();
 
     // save pet (hunter pet level and experience and all type pets health/mana).
-    if (Pet* pet = GetPet())
-        pet->SavePetToDB(PET_SAVE_AS_CURRENT);
+    if (getClass() != CLASS_PRIEST)
```

Comment:
well, it's nothing that's gonna make or break a PR, but it would be nice to have it unified rather than having the same check placed different places for the same purpose, also makes debugging easier should there be issues in relation to it.